core/execution.go

```diff
@@ -75,6 +79,101 @@ func Create(env vm.Environment, caller vm.ContractRef, code []byte, gas, gasPric
+	}
+	if !env.CanTransfer(caller.Address(), value) {
+		caller.ReturnGas(gas, gasPrice)
+
+		return nil, common.Address{}, ErrInsufficientFunds
+	}
+	nonce := env.Db().GetNonce(caller.Address())
+	env.Db().SetNonce(caller.Address(), nonce+1)
+
+	// Ensure there's no existing contract already at the designated address
+	// contractHash := evm.StateDB.GetCodeHash(address)
+	contractHash := env.Db().GetCodeHash(*address)
+	if env.Db().GetNonce(*address) != 0 || env.Db().GetCode(*address) != nil || (contractHash != (common.Hash{}) && contractHash != crypto.Keccak256Hash(nil)) {
+		return nil, common.Address{}, errContractAddressCollision
+	}
+	// Create a new account on the state
+	snapshot := env.SnapshotDatabase()
+
+	//Create account with address
+	to := env.Db().CreateAccount(*address)
+
+	if env.RuleSet().IsAtlantis(env.BlockNumber()) {
+		env.Db().SetNonce(*address, 1)
+	}
+
+	env.Transfer(env.Db().GetAccount(caller.Address()), env.Db().GetAccount(*address), value)
+
+	// Initialise a new contract and set the code that is to be used by the EVM.
+	// The contract is a scoped environment for this execution context only.
+	contract := vm.NewContract(caller, to, value, gas, gasPrice)
+	contract.SetCallCode(codeAddr, codeHash, code)
+	defer contract.Finalise()
+
+	ret, err := evm.Run(contract, input, readOnly)
+
+	// check whether the max code size has been exceeded
+	maxCodeSizeExceeded := len(ret) > maxCodeSize && env.RuleSet().IsAtlantis(env.BlockNumber())
+	// if the contract creation ran successfully and no errors were returned
+	// calculate the gas required to store the code. If the code could not
+	// be stored due to not enough gas set an error and let it be handled
+	// by the error checking condition below.
+	if err == nil && !maxCodeSizeExceeded {
+		// createDataGas := big.NewInt(int64(len(ret)) * int64(params.CreateDataGas))
+		createDataGas := big.NewInt(int64(len(ret)))
+		createDataGas.Mul(createDataGas, big.NewInt(200))
+		if contract.UseGas(createDataGas) {
+			env.Db().SetCode(*address, ret)
+		} else {
+			err = vm.CodeStoreOutOfGasError
+		}
+	}
+
+	// When an error was returned by the EVM or when setting the creation code
+	// above we revert to the snapshot and consume any gas remaining. Additionally
+	// when we're in homestead this also counts for code storage gas errors.
+	if maxCodeSizeExceeded || (err != nil && (env.RuleSet().IsHomestead(env.BlockNumber()) || err != errCodeStoreOutOfGas)) {
+		env.RevertToSnapshot(snapshot)
+		if err != vm.ErrRevert {
+			contract.UseGas(contract.Gas)
+		}
+	}
+	// Assign err if contract code size exceeds the max while the err is still empty.
+	if maxCodeSizeExceeded && err == nil {
+		err = errMaxCodeSizeExceeded
+	}
+
+	return ret, *address, err
+
+}
+
 func exec(env vm.Environment, caller vm.ContractRef, address, codeAddr *common.Address, codeHash common.Hash, input, code []byte, gas, gasPrice, value *big.Int, readOnly bool) (ret []byte, addr common.Address, err error) {
```

Comment:
@austinabell since create and create2 already have their own functions not relying on exec, would it make sense to just refactor execution.go so that the Calls are distinct as well for the sake of consistency?

Reply:
yeah, would be ideal but since there is no consensus issue (because of the explicit touching of STATICCALL in ETH) there is no immediate need to do this, but it would make a lot of sense to make these changes possibly in another PR? Seems like that would be out of the scope of this one

Reply:
I'll create a low priority issue